internal/command/deploy/machines.go

```diff
@@ -592,6 +592,11 @@ func (md *machineDeployment) resolveUpdatedMachineConfig(origMachineRaw *api.Mac
 		Config:  machineConf,
 		Region:  origMachineRaw.Region,
 	}
+
+	if launchInput.Region == "" {
+		launchInput.Region = md.appConfig.PrimaryRegion
```

Comment:
Region can be defaulted as part of origMachineRaw in line 580 https://github.com/superfly/flyctl/pull/1769/files#diff-4251d8215f73cf413b3febd989c9853c5c6d80d94c5b8ced61697ff05c8e563fR580